pkg/frontend/session.go

```diff
@@ -68,12 +69,26 @@ type TxnHandler struct {
 
 func InitTxnHandler(storage engine.Engine, txnClient TxnClient) *TxnHandler {
 	h := &TxnHandler{
-		storage:   storage,
+		storage:   &engine.EntireEngine{Engine: storage},
 		txnClient: txnClient,
 	}
 	return h
 }
 
+func (th *TxnHandler) SetTempEngine(te engine.Engine) {
+	th.mu.Lock()
```

Comment:
you lock with set but not lock when get?   Why is this useful?

actually, do you need to lock?   this is per session, right?    I don't know, what happens with remote execution and if you need remote read.

Reply:
Remote read doesn't use tempEngine in this session, however, we actually need to add lock for get.

Reply:
why?  single thread, why need a lock?

Reply:
> why? single thread, why need a lock?
Sorry I have dropped them but I forget to tell you.